### ISSUE TITLE
[Feature Request] Dimension Based Blacklist for Oil

### ISSUE THREAD

**Issue by MuteTiefling** [TeamPneumatic/pnc-repressurized#659]
Would you consider adding an additional dimension based blacklist for oil generation? It's a bit of a pain to track down all the nether/end biomes added by various mods and would be handy to simply blacklist the entire dimension instead.

**Comment by desht** [comment]
Yeah, that thought had occurred to me too.  I'll add that for the 2.7.0 release.

**Comment by desht** [comment]
This might actually be harder than I thought (and possibly why I didn't do it when I ported to 1.16.3) - the dimension ID simply isn't available to the `BiomeLoadingEvent`, which is where the oil lake feature is added.

I'll see if I can find a way, but it's not obvious at this point...

**Comment by MuteTiefling** [comment]
Oh, that's unfortunate to hear :(

**Comment by desht** [comment]
OK, here's a compromise: it's easy to blacklist by biome *category*, and both `nether` and `the_end` are valid categories.  So I'll add a "oil_worldgen_category_blacklist":

```
       #Oil worldgen blacklist by biome category: add biome categories to this list if you don't want oil worldgen to happen there. Accepted categories are: beach, desert, extreme_hills, forest, icy, jungle, mesa, mushroom, nether, none, ocean, plains, river, savanna, swamp, taiga, the_end.  This works in conjunction with 'oil_world_gen_blacklist' - if a biome matches either, then no oil lakes will generate there.
        oil_world_gen_category_blacklist = ["none"]
       #Oil worldgen blacklist: add biome IDs to this list if you don't want oil worldgen to happen there.  This works in conjunction with 'oil_world_gen_category_blacklist' - if a biome matches either, then no oil lakes will generate there.
        oil_world_gen_blacklist = ["minecraft:soul_sand_valley", "minecraft:crimson_forest", "minecraft:warped_forest", "minecraft:the_void", "minecraft:the_end", "minecraft:small_end_islands", "minecraft:end_midlands", "minecraft:end_highlands", "minecraft:end_barrens"]
```

So if you want no oil generating in *any* nether-type biome, adding "nether" to "oil_world_gen_category_blacklist" should do it, assuming the mod author has correctly categorised their new biome...

**Comment by MuteTiefling** [comment]
> assuming the mod author has correctly categorised their new biome...

Hey, you can only do so much from your end :D This is perfect, I think, and gets the job done. Thank you kindly!

**Comment by desht** [comment]
Fixed in 2.7.0 release
